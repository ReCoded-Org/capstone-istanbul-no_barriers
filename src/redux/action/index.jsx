import { USER_LOGGED_IN, WORDS_SEARCHED, LANGUAGE } from "./types";

export const userLoggedInAction = (data) => {
  return {
    type: USER_LOGGED_IN,
    payload: data,
  };
};

export const wordsSearchedAction = (data) => {
  return {
    type: WORDS_SEARCHED,
    payload: data,
  };
};

export const languageAction = (data) => {
  return {
    type: LANGUAGE,
    payload: data,
  };
};
