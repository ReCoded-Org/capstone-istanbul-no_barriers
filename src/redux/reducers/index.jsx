import { USER_LOGGED_IN } from "../action/types";
import { combineReducers } from "redux";

const currentUserDataReducer = (state = null, action) => {
  switch (action.type) {
    case USER_LOGGED_IN:
      return action.payload;
    default:
      return state;
  }
};

const allReducers = combineReducers({
  currentUserDataReducer,
});

export default allReducers;
