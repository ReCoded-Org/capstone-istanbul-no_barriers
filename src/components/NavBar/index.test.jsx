import React from "react";
import renderer from "react-test-renderer";
import { BrowserRouter as Router } from "react-router-dom";
import { Provider } from "react-redux";
import { createStore } from "redux";
import allReducers from "../../redux/reducers";

import Navbar from "./index";
import { Provider } from "react-redux";
import store from "../../redux/store/store";

const store = createStore(
  allReducers,
  window.devToolsExtension ? window.devToolsExtension() : (f) => f
);

it("navbar renders correctly", () => {
  const tree = renderer
    .create(
      <Provider store={store}>
        <Router>
          <Navbar />
        </Router>
      </Provider>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
