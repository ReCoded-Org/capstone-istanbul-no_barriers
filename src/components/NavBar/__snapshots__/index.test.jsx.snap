// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`navbar renders correctly 1`] = `
<nav
  className="navbar navbar-expand navbar-light"
>
  <a
    aria-current="page"
    className="navbar-brand active"
    href="/"
    onClick={[Function]}
    style={Object {}}
  >
    <span
      className="navbar-brand"
    >
      homePage.navbar.home
    </span>
  </a>
  <div
    className="ml-auto navbar-nav"
    onKeyDown={[Function]}
  >
    <a
      aria-current={null}
      className="nav-link"
      href="/findNgos"
      onClick={[Function]}
    >
      homePage.navbar.ngo
    </a>
    <a
      aria-current={null}
      className="nav-link"
      href="/stories"
      onClick={[Function]}
    >
      homePage.navbar.stories
    </a>
    <a
      aria-current={null}
      className="nav-link"
      href="/aboutUs"
      onClick={[Function]}
    >
      homePage.navbar.aboutUs
    </a>
    <a
      aria-current={null}
      className="nav-link"
      href="/profile"
      onClick={[Function]}
    >
      homePage.navbar.profile
    </a>
    <a
      aria-current={null}
      className="nav-link"
      href="/signin"
      onClick={[Function]}
    >
      <button
        className="shadow-none mr-1 btn btn-outline-info"
        disabled={false}
        type="button"
      >
        homePage.navbar.signIn
      </button>
    </a>
    <a
      aria-current={null}
      className="nav-link"
      href="/signup"
      onClick={[Function]}
    >
      <button
        className="shadow-none mr-1 btn btn-outline-info"
        disabled={false}
        type="button"
      >
        homePage.navbar.signUp
      </button>
    </a>
    <div
      className="dropleft nav-item"
      onKeyDown={[Function]}
    >
      <a
        aria-expanded={false}
        aria-haspopup={true}
        className="dropdown-toggle nav-link"
        data-rb-event-key={null}
        href="#"
        id="basic-nav-dropdown"
        onClick={[Function]}
        onKeyDown={[Function]}
        role="button"
      >
        homePage.navbar.dropdownTitle
      </a>
    </div>
  </div>
</nav>
`;
